x-pack/platform/plugins/shared/alerting/server/alerts_client/alerts_client.ts

```diff
@@ -679,6 +689,68 @@ export class AlertsClient<
     }
   }
 
+  public async updatePersistedAlerts() {
+    const { rawActiveAlerts } = this.getRawAlertInstancesForState(true);
+
+    const updateByQueryParams = Object.values(rawActiveAlerts).reduce(
+      (acc: UpdatePersistedAlertsQueryParams, rawAlert) => {
+        const { meta } = rawAlert;
+        if (!!meta?.uuid && meta.lastScheduledActions) {
+          const { group, date, actions } = meta.lastScheduledActions;
+          acc[meta.uuid] = {
+            group,
+            date,
+            ...(actions ? { actions } : {}),
+          };
+        }
+
+        return acc;
+      },
+      {}
+    );
+
+    const idsToUpdate = Object.keys(updateByQueryParams);
+
+    if (idsToUpdate.length === 0) {
+      return;
+    }
+    try {
+      const esClient = await this.options.elasticsearchClientPromise;
+      await retryTransientEsErrors(
+        () =>
+          esClient.updateByQuery({
+            query: {
+              terms: {
+                _id: idsToUpdate,
+              },
+            },
+            conflicts: 'proceed',
+            index: this.indexTemplateAndPattern.alias,
+            script: {
+              source: `
+            if (params.containsKey(ctx._source['${ALERT_UUID}'])) {
+              ctx._source['${ALERT_SCHEDULED_ACTION_GROUP}'] = params[ctx._source['${ALERT_UUID}']].group;
+              ctx._source['${ALERT_SCHEDULED_ACTION_DATE}'] = params[ctx._source['${ALERT_UUID}']].date;
+              if (params[ctx._source['${ALERT_UUID}']].containsKey('actions')) {
+                ctx._source['${ALERT_SCHEDULED_ACTION_THROTTLING}'] = params[ctx._source['${ALERT_UUID}']].actions;
+              }
+            }
+          `,
+              lang: 'painless',
+              params: updateByQueryParams,
+            },
+          }),
+        { logger: this.options.logger }
+      );
+    } catch (err) {
+      this.options.logger.error(
+        `Error updating alert last scheduled actions ${this.ruleInfoMessage}: ${err}`,
+        this.logTags
+      );
+      throw err;
```

Comment:
Do you think failing to update the alerts with the notification should fail the entire rule execution? For actions that run on every interval or on status change, this is primarily informational. I guess for throttled actions, it would lead to unexpected throttling behavior...

Reply:
`group` info is used to determine if action group has changed or not, missing this data would break `onStatusChange` actions as well. In the follow-on PR i will move `updateAlertsWithMaintenanceWindows` here too. missing all these may cause a lot of problem, IMO failing the execution is better, but we can discuss :)